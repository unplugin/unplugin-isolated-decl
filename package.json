{
  "name": "unplugin-isolated-decl",
  "version": "0.4.5",
  "packageManager": "pnpm@9.7.1",
  "description": "A blazing-fast tool for generating isolated declarations, powered by Oxc.",
  "type": "module",
  "keywords": [
    "unplugin",
    "rollup",
    "vite",
    "esbuild",
    "webpack"
  ],
  "license": "MIT",
  "homepage": "https://github.com/unplugin/unplugin-isolated-decl#readme",
  "bugs": {
    "url": "https://github.com/unplugin/unplugin-isolated-decl/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/unplugin/unplugin-isolated-decl.git"
  },
  "author": "三咲智子 <sxzz@sxzz.moe>",
  "files": [
    "dist"
  ],
  "main": "./dist/index.js",
  "module": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "require": "./dist/index.cjs",
      "import": "./dist/index.js"
    },
    "./vite": {
      "require": "./dist/vite.cjs",
      "import": "./dist/vite.js"
    },
    "./rollup": {
      "require": "./dist/rollup.cjs",
      "import": "./dist/rollup.js"
    },
    "./rolldown": {
      "require": "./dist/rolldown.cjs",
      "import": "./dist/rolldown.js"
    },
    "./esbuild": {
      "require": "./dist/esbuild.cjs",
      "import": "./dist/esbuild.js"
    },
    "./api": {
      "require": "./dist/api.cjs",
      "import": "./dist/api.js"
    },
    "./*": [
      "./*",
      "./*.d.ts"
    ]
  },
  "typesVersions": {
    "*": {
      "*": [
        "./dist/*",
        "./*"
      ]
    }
  },
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "lint": "eslint --cache .",
    "lint:fix": "pnpm run lint --fix",
    "build": "tsdown",
    "test": "vitest",
    "typecheck": "tsc --noEmit",
    "release": "bumpp && pnpm publish",
    "prepublishOnly": "pnpm run build"
  },
  "peerDependencies": {
    "@swc/core": "^1.6.6",
    "typescript": "^5.5.2"
  },
  "peerDependenciesMeta": {
    "@swc/core": {
      "optional": true
    },
    "typescript": {
      "optional": true
    }
  },
  "dependencies": {
    "@rollup/pluginutils": "^5.1.0",
    "oxc-transform": "^0.24.3",
    "unplugin": "^1.12.2"
  },
  "devDependencies": {
    "@swc/core": "^1.7.11",
    "@sxzz/eslint-config": "^3.17.2",
    "@sxzz/prettier-config": "^2.0.2",
    "@types/node": "^20.16.0",
    "bumpp": "^9.5.1",
    "esbuild": "^0.23.1",
    "eslint": "^9.9.0",
    "fast-glob": "^3.3.2",
    "prettier": "^3.3.3",
    "rollup": "^4.21.0",
    "tsdown": "^0.2.6",
    "tsx": "^4.17.0",
    "typescript": "^5.5.4",
    "vite": "^5.4.1",
    "vitest": "^2.0.5"
  },
  "engines": {
    "node": ">=18.12.0"
  },
  "prettier": "@sxzz/prettier-config"
}
