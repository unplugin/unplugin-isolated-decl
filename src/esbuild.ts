/**
 * This entry file is for esbuild plugin.
 *
 * @module
 */

import { IsolatedDecl } from './index'

/**
 * Esbuild plugin
 *
 * @example
 * ```ts
 * import { build } from 'esbuild'
 *
 * build({
 *   plugins: [require('unplugin-isolated-decl/esbuild')()],
 * })
 * ```
 */
const esbuild = IsolatedDecl.esbuild as typeof IsolatedDecl.esbuild
export default esbuild
export { esbuild as 'module.exports' }
