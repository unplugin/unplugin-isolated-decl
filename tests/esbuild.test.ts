import { readdir, readFile } from 'node:fs/promises'
import path from 'node:path'
import { build } from 'esbuild'
import { describe, expect, test } from 'vitest'
import { dependencies } from '../package.json'
import UnpluginIsolatedDecl from '../src/esbuild'

describe('esbuild', () => {
  const input = path.resolve(__dirname, 'fixtures/basic/main.ts')

  test('write mode', async () => {
    const dist = path.resolve(__dirname, 'temp/esbuild/basic')
    await build({
      entryPoints: [input],
      plugins: [UnpluginIsolatedDecl({ extraOutdir: 'extraOutdir' })],
      logLevel: 'silent',
      bundle: true,
      external: Object.keys(dependencies),
      platform: 'node',
      outdir: dist,
      format: 'esm',
    })

    const outDir = path.resolve(dist, 'extraOutdir')
    await expect(
      Promise.all(
        (await readdir(outDir))
          .sort()
          .map((file) => readFile(path.resolve(outDir, file), 'utf8')),
      ),
    ).resolves.toMatchSnapshot()
  })

  test('generate mode', async () => {
    const { outputFiles } = await build({
      entryPoints: [input],
      plugins: [UnpluginIsolatedDecl({ extraOutdir: 'temp' })],
      logLevel: 'silent',
      bundle: true,
      external: Object.keys(dependencies),
      platform: 'node',
      write: false,
      format: 'esm',
    })

    expect(
      outputFiles.map(
        (file) => `// ${file.path.replaceAll('\\', '/')}\n${file.text}`,
      ),
    ).toMatchSnapshot()
  })
})
