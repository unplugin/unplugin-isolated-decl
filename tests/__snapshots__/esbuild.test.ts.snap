// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`esbuild > generate mode 1`] = `
[
  "// <stdout>
// tests/fixtures/component.tsx
function Component() {
  return /* @__PURE__ */ React.createElement("div", null, "I'm a div in a tsx component!");
}

// tests/fixtures/main.ts
function hello(s) {
  return "hello" + s;
}
var c = Component;
var num = 1;
export {
  c,
  hello,
  num
};
",
  "// temp/main.d.ts
import { type Num } from './types';
export type Str = string;
export declare function hello(s: Str): Str;
export declare let c: React.JSX.Element;
export declare let num: Num;
",
  "// temp/types.d.ts
export type Num = number;
",
  "// temp/types2.d.ts
export type Num2 = number;
",
  "// temp/component.d.ts
export declare function Component(): React.JSX.Element;
",
]
`;

exports[`esbuild > write mode 1`] = `
[
  "export declare function Component(): React.JSX.Element;
",
  "import { type Num } from './types';
export type Str = string;
export declare function hello(s: Str): Str;
export declare let c: React.JSX.Element;
export declare let num: Num;
",
  "export type Num = number;
",
  "export type Num2 = number;
",
]
`;
